packages/flutter/lib/src/widgets/selection.dart

```diff
@@ -0,0 +1,586 @@
+class StaticMultiSelectableSelectionUpdater extends MultiSelectableSelectionUpdaterBase {
+  @override
+  List<Selectable> get selectables => _selectables;
+  List<Selectable> _selectables = const <Selectable>[];
+
+  DragSelectionStartEvent? _currentDragStart;
+
+  /// Updates the selectables this updater manages.
+  ///
+  /// This method will reset the ongoing selection.
+  void updateSelectables(List<Selectable> other) {
+    if (_selectables == other) {
+      return;
+    }
+    if (!listEquals<Selectable>(_selectables, other)) {
+      // Make sure the existing selectable finishes the drag selection
+      if (_currentDragStart != null) {
+        for (final Selectable selectable in _selectables) {
+          dispatchSelectionEventToChild(selectable, const DragSelectionEndEvent());
+        }
+      }
+      clear();
+      if (_currentDragStart != null) {
+        for (final Selectable selectable in _selectables) {
+          dispatchSelectionEventToChild(selectable, _currentDragStart!);
+        }
+      }
+    }
+    _selectables = other;
+  }
+
+  @override
+  SelectionResult dispatchSelectionEvent(SelectionEvent event) {
+    if (event is DragSelectionStartEvent) {
+      _currentDragStart = event;
+    } else if (event is DragSelectionEndEvent) {
+      _currentDragStart = null;
+    }
+    return super.dispatchSelectionEvent(event);
+  }
+
+}
+
+/// A container that collects the [Selectable]s in the subtree.
+///
+/// This widget creates its own [SelectionContainer] to collect the
+/// selectables in the subtree. These selectables are not visible to any other
+/// [SelectionContainer] above this widget. The creator of this widget
+/// can extract the selectables this container collected using the
+/// [onSelectablesChange] callback and is responsible for the selection update
+/// of the collected selectables.
+///
+/// If the [enabled] is false, this container does not collect selectables in
+/// the subtree, it will also hide them from other [SelectionContainer]
+/// above this widget.
+class SelectionContainer extends StatefulWidget {
+  /// creates a selection container to collect the [Selectable]s in the subtree.
+  ///
+  /// If the [enabled] is true, the [onSelectablesChange] must not be null.
+  const SelectionContainer({
+    Key? key,
+    required this.enabled,
+    this.onSelectablesChange,
+    required this.child
+  }) : assert(!enabled || onSelectablesChange != null),
+       super(key: key);
+
+  /// Whether this selection container is enabled.
+  ///
+  /// If this property is set to false, this container not only ignores all of
+  /// the selectables in the subtree, but it also hides them from any other
+  /// [SelectionContainer] above this widget.
+  final bool enabled;
+
+  /// The child widget this selection container contains.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  /// Called when the selectables this widget collected has changed.
```

Comment:
What constitutes a change? Is that just a selectable getting added or removed? Something else?